ase/sw/app_backend.c

```diff
@@ -1231,12 +1234,7 @@ void allocate_buffer(struct buffer_t *mem, uint64_t *suggested_vaddr)
     ase_buffer_info(mem);
 #endif
 
-    // book-keeping WSmeta // used by ASEALIAFU
-    struct wsmeta_t *ws;
-    ws = (struct wsmeta_t *) ase_malloc(sizeof(struct wsmeta_t));
-    ws->index = mem->index;
-    ws->buf_structaddr = (uint64_t *) mem;
-    append_wsmeta(ws);
+    append_buf(mem);
```

Comment:
This looks like you're tracking "mem" directly in a list, instead of allocating a new `wsmeta_t` entry referencing it. Are they equivalent?

Reply:
Yes, it is equivalent.